<launch>
  <!-- Arguments -->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />
  <arg name="move_forward_only" default="false"/>

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

    <rosparam file="$(find my_local_planner)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find my_local_planner)/param/costmap_common_params_$(arg model).yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find my_local_planner)/param/costmap_common_params_$(arg model).yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find my_local_planner)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find my_local_planner)/param/global_costmap_params.yaml" command="load" />

    <param name="base_global_planner" value="navfn/NavfnROS"/>
      <rosparam file="$(find my_local_planner)/param/navfnros_global_planner.yaml" command="load" />

    <param name="base_local_planner" value="my_local_planner/MyLocalPlanner" />
    
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    
  </node>
</launch>
